src/client/src/apps/MainRoute/widgets/spotTile/components/PriceControls/PriceControls.tsx

```diff
@@ -47,71 +47,57 @@ const PriceControls: React.FC<Props> = ({
     isRfqStateNone,
   } = getConstsFromRfqState(rfqState)
 
-  const isDisabled = !isRfqStateReceived && disabled
   const hasPrice = Boolean(priceData.bid && priceData.ask)
+  const showPrices =
+    isRfqStateNone || isRfqStateReceived || isRfqStateExpired || isTradeExecutionInFlight
+
+  const showPriceButton = (
+    btnDirection: Direction,
+    price: number,
+    rate: ReturnType<typeof toRate>,
+  ) => {
+    return showPrices ? (
+      <PriceButton
+        handleClick={() => executeTrade(btnDirection, price)}
+        direction={btnDirection}
+        big={rate.bigFigure}
+        pip={rate.pips}
+        tenth={rate.pipFraction}
+        rawRate={rate.rawRate}
+        priceAnnounced={isRfqStateReceived}
+        disabled={disabled}
+        expired={isRfqStateExpired}
+        currencyPairSymbol={currencyPair.symbol}
+      />
+    ) : null
+  }
+
+  if (!showPrices) {
+    var priceButtonDisabledStatus = isRfqStateCanRequest ? (
```

Comment:
Ouch, I've not seen a `var` in the wild for a long time!

Could we get rid of it by removing the `if` block and moving the `showPrices` logic into the ternary?

Reply:
sure, I'll move the `!showPrices` logic into both of the ternaries, but I personally don't see the issue with using `var` in this particular case. Is it a problem with consistency?

Reply:
Personally I avoid ever using it because of its "surprising" hoisting and scoping behaviour. You've used that behaviour here but it's not obvious

I'm a bit surprised we don't lint against it...